### PR TITLE
Wesley - Expose commit info

### LINKED ISSUE #13: FEATURE - expose commit infomation at /api/systemInfo
# User Story

* As a developer
* I can see information about the latest commit deployed to a deployment on dokku or localhost
* So that I am more aware of which commit is deployed on running code

### Discussion

In this issue, we bring over some code from proj-courses into this code base; code that provides transparency about what version of the code is running in any given deployment.

In the proj-courses code base, we've implemented a feature where you can get the SHA and the commit message for the latest commit that was deployed either on localhost or on dokku.  You access it by going to the endpoint /api/systemInfo.  

Here's what it looks like:

image

Note that this only works if all of the code is deployed correctly in the code base, *and* if the following has been done on the dokku deployment (here, `courses-qa` is the name of the dokku app)

```
  dokku git:set courses-qa keep-git-dir true
```

After setting that, you will need to do a `dokku ps:rebuild appname` for it to take effect.

If that's not set, you'll see something like this:

image

Also, you need to set the env variable: SOURCE_REPO to the correct repo.  This should be done rather than changing the value https://github.com/ucsb-cs156/proj-happycows in the code!

```
dokku config:set gauchoride  SOURCE_REPO=https://ucsb-cs156-s24/proj-gauchoride-s24-5pm-5
```

For the changes you need to make to the code, refer to this PR from proj-courses:

https://github.com/ucsb-cs156/proj-courses/pull/85

### Acceptance Criteria

- [ ] On Dokku, and on localhost, the /api/systemInfo shows the correct commit info about the most recent commit deployed to that deployment

### Implementation Todos

See: https://github.com/ucsb-cs156/proj-courses/pull/85 for hints on how to proceed.

### PR DESCRIPTION
**Pull request by jeffsmithepic**
Closes #13 

Added changes to the pom.xml, Dockerfile, and SystemInfo impl to allow for retrieving recent commit information through /api/systemInfo.

Admins can navigate to /api/systemInfo to find information about the most recent commit to the repository.

Steps to verify:
1. Log in at [Dev deployment](https://project-jeffsmithepic.dokku-14.cs.ucsb.edu/) as an admin user
2. Navigate to /api/systemInfo
3. Verify that recent commit info appears as in below screenshot

Image:
image

On a dokku deployment, the following commands need to be run first:
```
dokku git:set [app-name] keep-git-dir true
dokku config:set [app-name] SOURCE_REPO=https://ucsb-cs156-s24/proj-gauchoride-s24-5pm-6
```

[Dev deployment](https://project-jeffsmithepic.dokku-14.cs.ucsb.edu/)

Tests: Passing

### CODE REVIEW

**Review by lindseywn — DISMISSED** [review]
LGTM, working on dokku and tests passing

File: src/main/java/edu/ucsb/cs156/gauchoride/services/SystemInfoServiceImpl.java [hunk 1]
```diff
@@ -22,13 +27,39 @@ public class SystemInfoServiceImpl extends SystemInfoService {
   @Value("${app.showSwaggerUILink:false}")
   private boolean showSwaggerUILink;
 
+  @Value("${app.startQtrYYYYQ:20242}")
+  private String startQtrYYYYQ;
+
+  @Value("${app.endQtrYYYYQ:20243}")
+  private String endQtrYYYYQ;
+
+  @Value("${app.sourceRepo:https://github.com/ucsb-cs156/proj-gauchoride-s24-5pm-6}")
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/services/SystemInfoServiceImpl.java]: The default should be this, without the team number.
```suggestion
  @Value("${app.sourceRepo:https://github.com/ucsb-cs156/proj-gauchoride}")
```

The reason is that this code needs to be mergeable back in to the main repo that holds the production code.

**Review by pconrad — CHANGES_REQUESTED** [review]
One change we need to make for this to be mergeable...

File: src/main/java/edu/ucsb/cs156/gauchoride/services/SystemInfoServiceImpl.java (hunk 1 above)
Comment (jeffsmithepic) [reply src/main/java/edu/ucsb/cs156/gauchoride/services/SystemInfoServiceImpl.java]: Resolved in https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-6/commit/f54b7a3ee56433c9b984e89e73de111f210ab586, thanks!

**Review by yuji-sakaguchi — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**